interface conversion: interface {} is interp.valueInterface, not string when indexing map

### The following program `sample.go` triggers an unexpected result

```go
package main

import (
        "fmt"
)

func main() {
        var m = map[string]interface{}{"a": "a"}
        a, _ := m["a"]
        b := a.(string)
        fmt.Println(a, b)
}
```


### Expected result

```console
$ go run ./sample.go
a a
```


### Got

```console
$ yaegi ./sample.go
./sample.go:8:33: panic
run: interface conversion: interface {} is interp.valueInterface, not string
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc0002ea800, 0x40)
        /usr/local/go/src/runtime/debug/stack.go:24 +0x9f
github.com/traefik/yaegi/interp.(*Interpreter).Execute.func1(0xc0001adaf8)
        /go/src/yaegi/interp/program.go:131 +0xc8
panic(0xe43620, 0xc00019efb0)
        /usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/traefik/yaegi/interp.runCfg.func1(0xc0000cabb0, 0xc0004078c0, 0x0)
        /go/src/yaegi/interp/run.go:193 +0x251
panic(0xe43620, 0xc00019efb0)
        /usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/traefik/yaegi/interp.typeAssert.func3(0xc0000cabb0, 0x0)
        /go/src/yaegi/interp/run.go:441 +0x76b
github.com/traefik/yaegi/interp.runCfg(0xc0004078c0, 0xc0000cabb0, 0xc0004067e0, 0x0)
        /go/src/yaegi/interp/run.go:201 +0xbe
github.com/traefik/yaegi/interp.(*Interpreter).run(0xc0000ed8c0, 0xc0004067e0, 0xc0000caa50)
        /go/src/yaegi/interp/run.go:120 +0x2d2
github.com/traefik/yaegi/interp.(*Interpreter).Execute(0xc0000ed8c0, 0xc00048c9c0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/yaegi/interp/program.go:157 +0x245
github.com/traefik/yaegi/interp.(*Interpreter).eval(0xc0000ed8c0, 0xc0002d4fa0, 0x91, 0x7ffe1ea59eb4, 0xb, 0x0, 0x92, 0x0, 0x0, 0x91, ...)
        /go/src/yaegi/interp/interp.go:615 +0xd3
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0xc0000ed8c0, 0x7ffe1ea59eb4, 0xb, 0xc0002d4e00, 0x91, 0x0, 0x0, 0xc0001adc08)
        /go/src/yaegi/interp/interp.go:508 +0x13a
main.runFile(0xc0000ed8c0, 0x7ffe1ea59eb4, 0xb, 0x0, 0x1, 0x0)
        /go/src/yaegi/cmd/yaegi/run.go:153 +0xb1
main.run(0xc000032050, 0x1, 0x1, 0xc00000a501, 0xc000001680)
        /go/src/yaegi/cmd/yaegi/run.go:116 +0xa2a
main.main()
        /go/src/yaegi/cmd/yaegi/yaegi.go:144 +0x415
```


### Yaegi Version

fbee2baf9ddfad88f1d9b8c32fc3ca492e5c1ea5

### Additional Notes

The following patch seems to fix the problem (at least for this instance):

```diff
diff --git a/interp/run.go b/interp/run.go
index 1295807..354f976 100644
--- a/interp/run.go
+++ b/interp/run.go
@@ -1843,11 +1843,7 @@ func getIndexMap2(n *node) {
                        n.exec = func(f *frame) bltn {
                                v := value0(f).MapIndex(mi)
                                if v.IsValid() {
-                                       if e := v.Elem(); e.Type().AssignableTo(valueInterfaceType) {
-                                               dest(f).Set(e)
-                                       } else {
-                                               dest(f).Set(reflect.ValueOf(valueInterface{n, e}))
-                                       }
+                                       dest(f).Set(v.Elem())
                                }
                                if doStatus {
                                        value2(f).SetBool(v.IsValid())
@@ -1879,11 +1875,7 @@ func getIndexMap2(n *node) {
                        n.exec = func(f *frame) bltn {
                                v := value0(f).MapIndex(value1(f))
                                if v.IsValid() {
-                                       if e := v.Elem(); e.Type().AssignableTo(valueInterfaceType) {
-                                               dest(f).Set(e)
-                                       } else {
-                                               dest(f).Set(reflect.ValueOf(valueInterface{n, e}))
-                                       }
+                                       dest(f).Set(v.Elem())
                                }
                                if doStatus {
                                        value2(f).SetBool(v.IsValid())
```

But note that this is more trial and error than me really understanding what's going on.

--- a/_test/issue-1344.go
+++ b/_test/issue-1344.go
@@ -1,0 +1,13 @@
+package main
+
+import "fmt"
+
+func main() {
+	var m = map[string]interface{}{"a": "a"}
+	a, _ := m["a"]
+	b, ok := a.(string)
+	fmt.Println("a:", a, ", b:", b, ", ok:", ok)
+}
+
+// Output:
+// a: a , b: a , ok: true

--- a/interp/run.go
+++ b/interp/run.go
@@ -1766,9 +1766,6 @@ func getIndexArray(n *node) {
 	}
 }
 
-// valueInterfaceType is the reflection type of valueInterface.
-var valueInterfaceType = reflect.TypeOf((*valueInterface)(nil)).Elem()
-
 // getIndexMap retrieves map value from index.
 func getIndexMap(n *node) {
 	dest := genValue(n)

--- a/interp/run.go
+++ b/interp/run.go
@@ -1833,7 +1833,6 @@ func getIndexMap2(n *node) {
 	value0 := genValue(n.child[0])     // map
 	value2 := genValue(n.anc.child[1]) // status
 	next := getExec(n.tnext)
-	typ := n.anc.child[0].typ
 	doValue := n.anc.child[0].ident != "_"
 	doStatus := n.anc.child[1].ident != "_"
 
@@ -1848,21 +1847,6 @@ func getIndexMap2(n *node) {
 			n.exec = func(f *frame) bltn {
 				v := value0(f).MapIndex(mi)
 				value2(f).SetBool(v.IsValid())
-				return next
-			}
-		case isInterfaceSrc(typ):
-			n.exec = func(f *frame) bltn {
-				v := value0(f).MapIndex(mi)
-				if v.IsValid() {
-					if e := v.Elem(); e.Type().AssignableTo(valueInterfaceType) {
-						dest(f).Set(e)
-					} else {
-						dest(f).Set(reflect.ValueOf(valueInterface{n, e}))
-					}
-				}
-				if doStatus {
-					value2(f).SetBool(v.IsValid())
-				}
 				return next
 			}
 		default:
@@ -1884,21 +1868,6 @@ func getIndexMap2(n *node) {
 			n.exec = func(f *frame) bltn {
 				v := value0(f).MapIndex(value1(f))
 				value2(f).SetBool(v.IsValid())
-				return next
-			}
-		case isInterfaceSrc(typ):
-			n.exec = func(f *frame) bltn {
-				v := value0(f).MapIndex(value1(f))
-				if v.IsValid() {
-					if e := v.Elem(); e.Type().AssignableTo(valueInterfaceType) {
-						dest(f).Set(e)
-					} else {
-						dest(f).Set(reflect.ValueOf(valueInterface{n, e}))
-					}
-				}
-				if doStatus {
-					value2(f).SetBool(v.IsValid())
-				}
 				return next
 			}
 		default:

